nautobot/dcim/forms.py

```diff
@@ -1816,6 +1813,16 @@ class ComponentCreateForm(ComponentForm):
     description = forms.CharField(max_length=100, required=False)
 
 
+class ComponentEditForm(NautobotModelForm):
+    """
+    Base class for editing device components (models subclassed from ComponentModel).
+
+    Distinct from ComponentCreateForm in that it has a name/label instead of a name_pattern/label_pattern.
+    """
+
+    device = DynamicModelChoiceField(queryset=Device.objects.all())
```

Comment:
Could there be any negative impact of being able to change the device of an existing component?

Reply:
Yeah, it's not something that normally would/should be done. That said, I'm not sure it's necessarily *harmful* to permit it for that 1 in 1000 use case?

An alternate approach that I tried initially was making `device` be a hidden form field, but then I had to change the logic in `forms.js` that only permits population of `query_params` from visible fields, which I was also concerned about possible side effects from.

Thoughts? Back to the drawing board on this?

Reply:
We could disable that form field?

```py
    device = DynamicModelChoiceField(queryset=Device.objects.all(), disabled=True)
```

Reply:
Good thought. Had to do this conditionally in `__init__` because the "edit" form is actually used as the `model_form` for the "create" use case as well, so disabling it unconditionally would make component creation fail. But this appears to work.